scripts/core/driver.yml

```diff
@@ -271,3 +271,23 @@ members:
     - type: $x_driver_handle_t
       name: handle
       desc: "[in] handle of the driver instance"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Retrieves a message associated with the returned code of the last call made."
+version: "1.6"
+class: $xDriver
+name: GetLastResultString
```

Comment:
what's the goal of this function ?
how would this work in the driver?

Reply:
Maybe a better name would be `GetLastResultDescription`?

Reply:
Yeah I wonder how the string should be stored in the driver.
This would need to be dynamic memory allocation (i.e. string)
What about thread safety, if multiple calls are done in parallel ?
Does successful call free the string buffer?

After the API call, the string buffer needs to be still allocated or it can be freed?

Reply:
Would there be any opposition to these simply being static strings, similar to the `string_Vk...` routines in Vulkan SDK's `vk_enum_string_helper.h`?

Reply:
@MichalMrozek : interface is defined as thread-local, so this is expected to return a string describing the last returned code by that thread.

Right, I was thinking we would have a thread-local static array, and whenever the driver returns an error, like ZE_RESULT_UNKNOWN, driver will add a message to that static array. And on the next call, made by that that thread, the array is either cleaned (if we are returning ZE_RESULT_SUCCESS) or rewritten if we are returning a new error code.

To avoid having to write the array on SUCCESS calls, maybe we could add the provision that this returns a message associated with the last error code, and that message stays in the array until the next error code:

Calls that return SUCCESS dont overwrite the array or message.

Reply:
I would like to make sure that driver does not need to do anything on SUCCESS case.
SUCCESS case needs to be razor thin efficient.

Reply:
@MichalMrozek : right, that's a very good point and that aligns with expectation, since this is more for errors than for success. So @wdamon-intel  suggestion of calling it "GetLastErrorDescription" is good, and we could add something like:

```html
- String is updated only when driver returns an error code.
- Calling zeDriverGetLastError after a call that returns SUCCESS will retrieve the same string associated with last call that returned an error other than SUCCESS.
```

Reply:
addressed comments